packages/ledgerstate/branch.go

```diff
@@ -0,0 +1,981 @@
+		stringify.StructField("conflicts", b.Conflicts()),
+		stringify.StructField("preferred", b.Preferred()),
+		stringify.StructField("liked", b.Liked()),
+		stringify.StructField("finalized", b.Finalized()),
+		stringify.StructField("confirmed", b.Confirmed()),
+		stringify.StructField("rejected", b.Rejected()),
+	)
+}
+
+// Update is disabled and panics if it ever gets called - it is required to match the StorableObject interface.
+func (b *ConflictBranch) Update(objectstorage.StorableObject) {
+	panic("updates disabled")
+}
+
+// ObjectStorageKey returns the key that is used to store the object in the database. It is required to match the
+// StorableObject interface.
+func (b *ConflictBranch) ObjectStorageKey() []byte {
+	return b.ID().Bytes()
+}
+
+// ObjectStorageValue marshals the ConflictBranch into a sequence of bytes that are used as the value part in the
+// ObjectStorage.
+func (b *ConflictBranch) ObjectStorageValue() []byte {
+	return marshalutil.New().
+		WriteByte(byte(b.Type())).
+		WriteBytes(b.ID().Bytes()).
+		WriteBytes(b.Parents().Bytes()).
+		WriteBytes(b.Conflicts().Bytes()).
+		WriteBool(b.Preferred()).
+		WriteBool(b.Liked()).
+		WriteBool(b.Finalized()).
+		WriteBool(b.Confirmed()).
+		WriteBool(b.Rejected()).
+		Bytes()
+}
+
+// code contract (make sure the struct implements all required methods)
+var _ Branch = &ConflictBranch{}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region AggregatedBranch ///////////////////////////////////////////////////////////////////////////////////////////////
+
+// AggregatedBranch represents a container for Transactions and Outputs representing a certain perception of the ledger state.
+type AggregatedBranch struct {
+	id             BranchID
+	parents        BranchIDs
+	parentsMutex   sync.RWMutex
+	preferred      bool
+	preferredMutex sync.RWMutex
+	liked          bool
+	likedMutex     sync.RWMutex
+	finalized      bool
+	finalizedMutex sync.RWMutex
+	confirmed      bool
+	confirmedMutex sync.RWMutex
+	rejected       bool
+	rejectedMutex  sync.RWMutex
+
+	objectstorage.StorableObjectFlags
+}
+
+// NewAggregatedBranch creates a new AggregatedBranch from the given details.
+func NewAggregatedBranch(parents BranchIDs) *AggregatedBranch {
+	// sort parents
+	parentBranchIDs := parents.Slice()
+	sort.Slice(parentBranchIDs, func(i, j int) bool {
+		for k := 0; k < len(parentBranchIDs[k]); k++ {
+			if parentBranchIDs[i][k] < parentBranchIDs[j][k] {
+				return true
+			} else if parentBranchIDs[i][k] > parentBranchIDs[j][k] {
+				return false
+			}
+		}
+
+		return false
+	})
+
+	// concatenate sorted parent bytes
+	marshalUtil := marshalutil.New(BranchIDLength * len(parentBranchIDs))
+	for _, branchID := range parentBranchIDs {
+		marshalUtil.WriteBytes(branchID.Bytes())
+	}
+
+	// return result
+	return &AggregatedBranch{
+		id:      blake2b.Sum256(marshalUtil.Bytes()),
+		parents: parents,
```

Comment:
Here I think we should copy. Otherwise we need to make sure that `parents` are not used outside of the just created `AggregatedBranch `. This could create some nasty concurrency bugs.